lti_consumer/models.py

```diff
@@ -526,3 +528,58 @@ def __str__(self):
 
     class Meta:
         app_label = 'lti_consumer'
+
+
+class CourseEditLTIFieldsEnabledFlag(ConfigurationModel):
+    """
+    Enables the editing of "request username" and "request email" fields
+    of LTI consumer for a specific course.
+
+    .. no_pii:
+    """
+    KEY_FIELDS = ('course_id',)
+
+    course_id = CourseKeyField(max_length=255, db_index=True)
+
+    @classmethod
+    @request_cached
+    def lti_access_to_learners_editable(cls, course_id, is_already_sharing_learner_info):
+        """
+        Looks at the currently active configuration model to determine whether
+        the feature that enables editing of "request username" and "request email"
+        fields of LTI consumer is available or not.
+
+        Backwards Compatibility:
+        Enable this feature for a course run who was sharing learner username/email
+        in the past.
+
+        Arguments:
+            course_id (CourseKey): course id for which we need to check this configuration
+            is_already_sharing_learner_info (bool): indicates whether LTI consumer is
+            already sharing edX learner username/email.
+        """
+        course_specific_config = (CourseEditLTIFieldsEnabledFlag.objects
```

Comment:
Using the `ConfigurationModel` as baseclass, you should be able to 
```
CourseEditLTIFieldsEnabledFlag.objects.filter(course_id=course_id).current()
```
Is there specific reason why we not using `.current()`?

Reply:
I'm not sure! I just copied it over since I din't want to change too many things at once.

Reply:
I found out why this is the way it is. Calling `.current` automatically creates a new object if one doesn't exist. However in this case the object should only be created if `is_already_sharing_learner_info` is true, and not otherwise.

Reply:
thanks for checking @xitij2000

Reply:
thanks for checking @xitij2000